Prevent NaNs from emitting a warning in Latitude

dstansby opened the pull request:
In a similar vein to #9644 and #9598, this prevents a "invalid value" warning being thrown by numpy when a `Nan` is given to `astropy.coordinates.Latitude`.

mhvk commented:
@dstansby - numpy is fixing this upstream (e.g., https://github.com/numpy/numpy/pull/15230), so I wonder whether it doesn't make more sense to just let things be.

pllim commented:
> just let things be

Maybe we can still apply this logic for older versions of Numpy only, and then remove this logic when our Numpy minversion is high enough, like how we deal with some of the other stuff, @mhvk?

mhvk commented:
@pllim - good point, that helps keep the noise down for users and ensures that we remove it when no longer needed. Would just need to check under which numpy versions the warning is raised (where it helps that we only support >=1.16!). @dstansby - up to you if you want to do that...

dstansby commented:
Given the numpy PR only went in 2 days ago, I suspect it's not made it to a full release yet.

bsipocz commented:
@dstansby - could you rebase?

@mhvk - I still see the RuntimeWarnings with np 1.18.1, and as we keep supporting np 1.16, we might rather have this for now? 
On the other hand, both cases are fine I think, but it would be better to have  a resolution for the PR and either merge or close it rather than keeping it open.

mhvk commented:
I'm still a bit torn on this. Arguably, `Latitude` could also error on getting `nan` since that number neither is in nor outside of the valid range...

Anyway, maybe best is to just go with it, but then have at least be some comment about the versions of numpy with which this actually is a problem, so that, as @pllim notes, we can be sure we remove it when the issue has been resolved upstream. Ideally phrase this as something like `not NUMPY_LT_1_18_1` (or whatever) so that grepping for `NUMPY_LT` will catch the comment...

dstansby commented:
I've added comments to all of the `np.errstate` bits that I've added. Note that these should be pretty easy to find by searching for `np.errstate(invalid='ignore')` too.
